src/GameStateMenu.cpp

```diff
@@ -10,143 +10,509 @@
+        		if (x >= textboxPos[i].x && x <= textboxPos[i].x + textboxPos[i].w &&
+        			y >= textboxPos[i].y && y <= textboxPos[i].y + textboxPos[i].h) {
+                    activeTextbox[i] = true; //Activate the textbox to enable typing
+                    //If default string is detected, delete it
+                    if (!strcmp(textInput[i].c_str(), defaultText[i].c_str())) {
+                        textInput[i] = "";
+                    }
+                } else {
+                    activeTextbox[i] = false; //deactivate the textbox
+                    //Reset textbox text to default text, if textbox is empty
+                    if (textInput[i].empty()) {
+                        textInput[i] = defaultText[i];
+                    }
+        		}
+        	}
+
+        	break;
+        case SDL_WINDOWEVENT:
+        	switch (event.window.event) {
+	    		case SDL_WINDOWEVENT_RESIZED:
+                    //Adjust the dimensions of the window if resized
+	    			level.levelTexture.setDimensions(game.window.getWidth(),
+                                                    game.window.getHeight());
+	    			break;
+	        }
+        	break;
+		case SDL_QUIT:
+			play = false;
+			break;
+      	default:
+        	break;
+	}
 }
 
+/**
+* Function: update
+*
+* Date:
+* JF: February 8, 2017: Moved positional data into own method
+*
+* Designer:
+* Jacob Frank
+*
+* Programmer:
+* Jacob Frank
+*
+* Interface: update(const float& delta)
+*
+* Returns: void
+*
+* Notes:
+* Function positions all screen elements in the window
+*/
 void GameStateMenu::update(const float& delta) {
 
-    // TEMP: Skip to GameStateMatch
-    // Remove this when working on the main menu
-    game.stateID = 2;
-    play = false;
-
+	if(delta == JOIN) {
+		game.stateID = 2;
+	} else if (delta == OPTIONS) {
+        game.stateID = 2; //TEMPORARY: change to correct state ID once implemented
+	} else {
+        game.stateID = 0;
+	}
 }
 
-void GameStateMenu::render() {
-    //Only draw when not minimized
-    if ( !game.window.isMinimized() ) {
-
-        //Clear screen
-        SDL_SetRenderDrawColor( game.renderer, 0xFF, 0xFF, 0xFF, 0xFF );
-        SDL_RenderClear( game.renderer );
-
-        SDL_Color textColor = { 0, 0, 0, 255 };
+/**
+* Function: positionElements
+*
+* Date:
+* JF: March 6, 2017: Moved positional data into own method
+*
+* Designer:
+* Jacob Frank
+*
+* Programmer:
+* Jacob Frank
+*
+* Interface: positionElements()
+*
+* Returns: void
+*
+* Notes:
+* Function positions all screen elements in the window
+*/
+void GameStateMenu::positionElements() {
+
+    int         windowWidth     = game.window.getWidth();
+    int         windowHeight    = game.window.getHeight();
+    int         maxTextWidth    = 0;
+    int         maxTextHeight   = 0;
+    int         vertPadding     = 50;
+    int         horzPadding     = 10;
+    std::string longestString   = "";
+    char        largestChar     = 'W';
```

Comment:
this isnt a blocking issue but for the future a single space between variable and name as well as name and value will be added to the style guide.